src/bin/seed-database.rs

```diff
@@ -0,0 +1,207 @@
+        None => load_random_data(storage, generator_config),
+    }
+
+    Ok(())
+}
+
+/// Type for data format stored in mainnetMM data dump
+#[derive(Debug, Clone)]
+pub struct BlockRlp {
+    pub header: Option<Header>,
+}
+
+impl Decodable for BlockRlp {
+    fn decode(rlp: &rlp::Rlp) -> Result<Self, DecoderError> {
+        let header = match Header::decode_rlp(rlp) {
+            Ok(val) => Some(val),
+            Err(_) => None,
+        };
+        Ok(Self { header })
+    }
+}
+
+fn load_mainnet_data(mut storage: PortalStorage, data_folder: String) {
+    let mainnet_data_dir = Path::new(&data_folder);
+    let paths = fs::read_dir(mainnet_data_dir).unwrap();
+    for path in paths {
+        load_file_data(&path.unwrap().path(), &mut storage);
+    }
+}
+
+fn load_file_data(path: &Path, storage: &mut PortalStorage) {
+    println!("Loading data from file: {:?}", path);
+    let file = File::open(path).unwrap();
+    let reader = BufReader::new(file);
+    let json_blocks: Value = serde_json::from_reader(reader).unwrap();
+    let mut stored_count = 0;
+    let mut total_count = 0;
+    match json_blocks {
+        Value::Object(val) => {
+            for (_block_hash, block_data) in val.into_iter() {
+                match block_data {
+                    Value::Object(val) => {
+                        let block_rlp = val.get("rlp").unwrap().as_str().unwrap().to_string();
+                        let prefix_removed_rlp = &block_rlp[2..block_rlp.len()];
+                        let bytes_rlp = hex::decode(prefix_removed_rlp).unwrap();
+                        let blocks: Vec<BlockRlp> = rlp::decode_list(&bytes_rlp);
+                        let block = blocks.get(0).unwrap();
+
+                        if block.header.is_some() {
+                            let content_key = HistoryContentKey::BlockHeader(BlockHeader {
+                                chain_id: 1u16,
+                                block_hash: block.clone().header.unwrap().hash().into(),
+                            });
+
+                            let content_key_bytes: Vec<u8> = content_key.clone().into();
+                            let content_key_hex = hex::encode(&content_key_bytes);
+                            total_count += 1;
+                            match storage.should_store(&content_key).unwrap() {
+                                true => {
+                                    storage.store(&content_key, &bytes_rlp).unwrap();
+                                    println!("Stored content key: {:?}", content_key_hex);
+                                    println!("- Block RLP: {:?}", prefix_removed_rlp.get(0..31));
+                                    stored_count += 1;
+                                }
+                                false => {}
+                            }
+                        }
+                    }
+                    _ => panic!("Invalid deserialized type, expected an object."),
```

Comment:
What will happen if this fails in the middle of data loading? Should we think at this stage about idempotency and how we can resolve unexpected panics?

Reply:
Good point - updated this `panic!`s to `debug!`s